src/main/java/org/datadog/jenkins/plugins/datadog/DatadogUtilities.java

```diff
@@ -832,6 +832,33 @@ public static String toJson(final Set<String> set) {
         return sb.toString();
     }
 
+    /**
+     * Returns a JSON object string based on the map.
+     * @param map the map to transform into a JSON
+     * @return json object string
+     */
+    public static String toJson(final Map<String, String> map) {
+        if(map == null || map.isEmpty()) {
+            return "";
+        }
+
+        // We want to avoid using Json libraries cause
+        // may cause incompatibilities on different Jenkins versions.
+        final StringBuilder sb = new StringBuilder();
+        sb.append("{");
+        int index = 1;
+        for(Map.Entry<String, String> entry : map.entrySet()) {
+            sb.append("\"").append(entry.getKey()).append("\"").append(":").append("\"").append(entry.getValue()).append("\"");
```

Comment:
nit: is there a way to do this with sprintf formatting or something like that? It'd be easier to follow

Reply:
You should escape the key and value before adding them to the string. You might end up with invalid JSON if you don't do that.